ui/app/pages/swaps/swaps-gas-customization-modal/swaps-gas-customization-modal.container.js

```diff
@@ -121,18 +128,20 @@ const mapStateToProps = (state) => {
     customGasLimitMessage,
     customTotalSupplement,
     usdConversionRate: getUSDConversionRate(state),
-    disableSave: insufficientBalance || customGasLimit < minimumGasLimit,
+    disableSave: insufficientBalance || customGasLimitTooLow,
     minimumGasLimit,
   }
 }
 
 const mapDispatchToProps = (dispatch) => {
   return {
     cancelAndClose: () => {
+      dispatch(swapCustomGasModalClosed())
       dispatch(hideModal())
     },
-    onSubmit: (gasLimit, gasPrice) => {
-      dispatch(customSwapsGasParamsUpdated(gasLimit, gasPrice))
+    onSubmit: async (gasLimit, gasPrice) => {
+      await dispatch(customSwapsGasParamsUpdated(gasLimit, gasPrice))
+      dispatch(swapCustomGasModalClosed())
```

Comment:
Ensure that custom modal gas limit and price amounts are reset when saving the those amounts. The `customSwapsGasParamsUpdated` call ensures they get saved to the background, so that they are persisted and modal will properly populate next time it is opened.

Reply:
Just to confirm, that data comes from the quotes and are passed to the gas customization modal as props based on the selected quote?

Reply:
`gasPrice` comes from our gas estimation api

Initial (non-custom) gas limits come either from our backend gas estimation or from the `maxGas` property on the selected quote. If either have been edited in the modal and saved previously, and not cleared due to some other action, then the edited (custom) values are what are passed to the modal.